### PR TITLE
✅  🐛  🏗 Fix Percy flakes stemming from incomplete opacity transitions

### LINKED ISSUE #32461: Increased visual-diff flakiness during CI
After running `gulp visual-diff` on CI, we have the odd diff show up in Percy. Of late, I'm seeing way more flaky tests. Not sure if this is due to new tests, or if it's an infrastructure issue (new version of Chrome, puppeteer, or the Percy agent?) that's causing this spike in flakiness.

E.g. See [these diffs](https://percy.io/ampproject/amphtml/builds/8716818/changed/495171451?browser=firefox&browser_ids=5&subcategories=unreviewed%2Cchanges_requested%2Capproved&viewLayout=side-by-side&viewMode=new&width=800&widths=375%2C411%2C800%2C1400) for a PR that does not touch runtime code.

### PR DESCRIPTION
**Pull request by rileyajones**
Visual diffs sometimes fail due to incomplete opacity transitions.

fixes #32461

### CODE REVIEW

**Review by rsimha — COMMENTED** [review]
Thanks for working on this! Conceptually LGTM from me, with two notes:
1. Seems like one of your fork's branches are out of date? (the diffs in this PR contain a change from an old PR)
2. Let's have Daniel review / approve this change tomorrow and make sure there aren't other corner cases to consider

**Comment by rileyajones** [comment]
This will probably result in some percy failures but future tests should be more consistent

**Comment by rileyajones** [comment]
Rebased the fork

**Comment by danielrozenberg** [comment]
Do you have specific examples Percy builds that show this problem?

**Comment by rileyajones** [comment]
> Do you have specific examples Percy builds that show this problem?

Here's an example of a recent one of mine that failed this way
https://percy.io/ampproject/amphtml/builds/8662656

**Comment by rsimha** [comment]
A couple minor meta-comments (while @danielrozenberg continues to review the code):
- The PR linked in the description #32442 is about E2E tests, while this PR is about visual tests
- To auto-close a linked issue when the PR is merged, use [these strings](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (e.g. `Fixes #12345` or `Closes #12345`)

File: build-system/tasks/visual-diff/helpers.js [hunk 1]
```diff
@@ -191,8 +191,12 @@ async function waitForElementVisibility(page, selector, options) {
 
     for (const elementHandle of await page.$$(selector)) {
       const boundingBox = await elementHandle.boundingBox();
+      const fullyOpaque = (await getElementOpacity(elementHandle)) === '1';
```
Comment (dvoytenko) [inline build-system/tasks/visual-diff/helpers.js:194]: Sometimes an element can naturally have lower opacity in the document's stylesheet. I think this is generally ok - will just be a limitation of the diffing system. We just need to call it out in the docs.

Comment (rileyajones) [reply build-system/tasks/visual-diff/helpers.js:196]: That's a great point, we should definitely call it out in the docs.

Comment (danielrozenberg) [reply build-system/tasks/visual-diff/helpers.js:196]: I'm just still feeling like this is prone to come back and bite us eventually, so I wanna be extra careful here.
@rileyajones I assume this is meant to handle the case of animated transitions? In general Percy forces all animations to be 0 seconds long on server side, if this is only supposed to handle CSS transitions perhaps we can force the same kind of constraint on our visual diffs instead of awaiting on the opacity (basically, treat the cause and not the effect) - is this the case you're trying to handle?
